From 4904ce6a69032e6da5065603e397ec590d1b5350 Mon Sep 17 00:00:00 2001
From: Dimo <dimitry-ishenko@users.noreply.github.com>
Date: Tue, 3 Mar 2026 04:55:49 -0500
Subject: [PATCH] fix: Remove dependency on libpostproc (#1709)

It's not used anywhere and has been removed from ffmpeg 8.
---
 src/CMakeModules/Bootstrap_Windows.cmake    | 1 -
 src/CMakeModules/FindFFmpeg.cmake           | 6 ++----
 tools/linux/deb/ubuntu-noble/debian/control | 1 -
 tools/linux/install-dependencies            | 1 -
 4 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/src/CMakeModules/Bootstrap_Windows.cmake b/src/CMakeModules/Bootstrap_Windows.cmake
index 693148a93f..01652e13f9 100644
--- a/src/CMakeModules/Bootstrap_Windows.cmake
+++ b/src/CMakeModules/Bootstrap_Windows.cmake
@@ -106,7 +106,6 @@ casparcg_add_runtime_dependency("${FFMPEG_BIN_PATH}/avdevice-61.dll")
 casparcg_add_runtime_dependency("${FFMPEG_BIN_PATH}/avfilter-10.dll")
 casparcg_add_runtime_dependency("${FFMPEG_BIN_PATH}/avformat-61.dll")
 casparcg_add_runtime_dependency("${FFMPEG_BIN_PATH}/avutil-59.dll")
-casparcg_add_runtime_dependency("${FFMPEG_BIN_PATH}/postproc-58.dll")
 casparcg_add_runtime_dependency("${FFMPEG_BIN_PATH}/swresample-5.dll")
 casparcg_add_runtime_dependency("${FFMPEG_BIN_PATH}/swscale-8.dll")
 # for scanner:
diff --git a/src/CMakeModules/FindFFmpeg.cmake b/src/CMakeModules/FindFFmpeg.cmake
index 87503f226a..4ebaa2ee4c 100644
--- a/src/CMakeModules/FindFFmpeg.cmake
+++ b/src/CMakeModules/FindFFmpeg.cmake
@@ -13,7 +13,6 @@
 #   - AVFORMAT
 #   - AVFILTER
 #   - AVUTIL
-#   - POSTPROC
 #   - SWSCALE
 # the following variables will be defined
 #  <component>_FOUND        - System has <component>
@@ -33,7 +32,7 @@ include(FindPackageHandleStandardArgs)
 
 # The default components were taken from a survey over other FindFFMPEG.cmake files
 if (NOT FFmpeg_FIND_COMPONENTS)
-  set(FFmpeg_FIND_COMPONENTS AVCODEC AVFORMAT AVUTIL AVDEVICE AVFILTER POSTPROC SWSCALE SWRESAMPLE)
+  set(FFmpeg_FIND_COMPONENTS AVCODEC AVFORMAT AVUTIL AVDEVICE AVFILTER SWSCALE SWRESAMPLE)
 endif ()
 
 #
@@ -107,7 +106,6 @@ if (NOT FFMPEG_LIBRARIES)
   find_component(AVUTIL     libavutil     avutil   libavutil/avutil.h)
   find_component(AVFILTER   libavfilter   avfilter libavfilter/avfilter.h)
   find_component(SWSCALE    libswscale    swscale  libswscale/swscale.h)
-  find_component(POSTPROC   libpostproc   postproc libpostproc/postprocess.h)
   find_component(SWRESAMPLE libswresample swresample libswresample/swresample.h)
 
   # Check if the required components were found and add their stuff to the FFMPEG_* vars.
@@ -149,7 +147,7 @@ if (NOT FFMPEG_LIBRARIES)
 endif ()
 
 # Now set the noncached _FOUND vars for the components.
-foreach (_component AVCODEC AVDEVICE AVFORMAT AVUTIL POSTPROCESS SWSCALE)
+foreach (_component AVCODEC AVDEVICE AVFORMAT AVUTIL SWSCALE)
   set_component_found(${_component})
 endforeach ()
 
diff --git a/tools/linux/deb/ubuntu-noble/debian/control b/tools/linux/deb/ubuntu-noble/debian/control
index 9c762d961f..9ed0118af7 100644
--- a/tools/linux/deb/ubuntu-noble/debian/control
+++ b/tools/linux/deb/ubuntu-noble/debian/control
@@ -24,7 +24,6 @@ Build-Depends: cmake,
                libglew-dev,
                libjpeg-dev,
                libopenal-dev,
-               libpostproc-dev,
                libsfml-dev,
                libsimde-dev,
                libswresample-dev,
diff --git a/tools/linux/install-dependencies b/tools/linux/install-dependencies
index 81e443262d..ebf41e2596 100755
--- a/tools/linux/install-dependencies
+++ b/tools/linux/install-dependencies
@@ -46,6 +46,5 @@ apt-get install -yq --no-install-recommends \
   libavutil-dev \
   libavfilter-dev \
   libswscale-dev \
-  libpostproc-dev \
   libswresample-dev \
   libsimde-dev
